proto/FeatureCollection/FeatureCollection.proto

```diff
@@ -1,210 +1,256 @@
-    UniqueIdField uniqueIdField = 2;
-    string globalIdFieldName = 3;
-    string geohashFieldName = 4;
-    GeometryProperties geometryProperties = 5;
-    ServerGens serverGens = 6;
-    GeometryType geometryType = 7;
-    SpatialReference spatialReference = 8;
-    bool exceededTransferLimit = 9;
-    bool hasZ = 10;
-    bool hasM = 11;
-    Transform transform = 12;
-    repeated Field fields = 13;
-    repeated Value values = 14;
-    repeated Feature features = 15;
-  }
-
-  message CountResult{
-    uint64 count = 1;
-  }
-
-  message ObjectIdsResult {
-    string objectIdFieldName = 1;
-    ServerGens serverGens = 2;
-    repeated uint64 objectIds = 3 [packed = true];
-  }
-
-  message QueryResult {
-    oneof Results {
-      FeatureResult featureResult = 1;
-      CountResult countResult = 2;
-      ObjectIdsResult idsResult = 3;
-    }
-  }
-
-  // Any compliant implementation must first read the version
-  // number encoded in this message and choose the correct
-  // implementation for this version number before proceeding to
-  // decode other parts of this message.
-  string version = 1;
-  QueryResult queryResult = 2;
+		// GeometryType
+		enum GeometryType {
+			esriGeometryTypePoint = 0;
+			esriGeometryTypeMultipoint = 1;
+			esriGeometryTypePolyline = 2;
+			esriGeometryTypePolygon = 3;
+			esriGeometryTypeMultipatch = 4;
+			esriGeometryTypeNone = 127;
+			esriGeometryTypeEnvelope = 5;
+		}
+
+		// FieldType
+		enum FieldType {
+			esriFieldTypeSmallInteger = 0;
+			esriFieldTypeInteger = 1;
+			esriFieldTypeSingle = 2;
+			esriFieldTypeDouble = 3;
+			esriFieldTypeString = 4;
+			esriFieldTypeDate = 5;
+			esriFieldTypeOID = 6;
+			esriFieldTypeGeometry = 7;
+			esriFieldTypeBlob = 8;
+			esriFieldTypeRaster = 9;
+			esriFieldTypeGUID = 10;
+			esriFieldTypeGlobalID = 11;
+			esriFieldTypeXML = 12;
+			esriFieldTypeBigInteger = 13;
+			esriFieldTypeDateOnly = 14;
+			esriFieldTypeTimeOnly = 15;
+			esriFieldTypeTimestampOffset = 16;
+		}
+
+		// FieldType
+		enum SQLType {
+			sqlTypeBigInt = 0;
+			sqlTypeBinary = 1;
+			sqlTypeBit = 2;
+			sqlTypeChar = 3;
+			sqlTypeDate = 4;
+			sqlTypeDecimal = 5;
+			sqlTypeDouble = 6;
+			sqlTypeFloat = 7;
+			sqlTypeGeometry = 8;
+			sqlTypeGUID = 9;
+			sqlTypeInteger = 10;
+			sqlTypeLongNVarchar = 11;
+			sqlTypeLongVarbinary = 12;
+			sqlTypeLongVarchar = 13;
+			sqlTypeNChar = 14;
+			sqlTypeNVarchar = 15;
+			sqlTypeOther = 16;
+			sqlTypeReal = 17;
+			sqlTypeSmallInt = 18;
+			sqlTypeSqlXml = 19;
+			sqlTypeTime = 20;
+			sqlTypeTimestamp = 21;
+			sqlTypeTimestamp2 = 22;
+			sqlTypeTinyInt = 23;
+			sqlTypeVarbinary = 24;
+			sqlTypeVarchar = 25;
+			sqlTypeTimestampWithTimezone = 26;
+		}
+
+		enum QuantizeOriginPostion {
+			upperLeft = 0;
+			lowerLeft = 1;
+		}
+
+		// SegmentType
+		enum SegmentType {
+			line = 0;
+			arc = 1;
+			bezier = 2;
+			ellipticArc = 3;
+		}
+
+		message SpatialReference {
+			uint32 wkid = 1;
+			uint32 lastestWkid = 2;

```

Comment:
Please note that this is a bug. It should be `latestWkid` there is a PR from last year that fixed this https://github.com/Esri/arcgis-pbf/pull/9 that is unmerged

Reply:
This should really be fixed upstream, not here. We don't want this version to be out of sync with what's used internally.